typescript/packages/wallets/fuel/src/FuelWalletClient.ts

```diff
@@ -0,0 +1,76 @@
+import { WalletClientBase } from "@goat-sdk/core";
+import { bn, formatUnits, TransactionRequest, WalletUnlocked } from "fuels";
+
+export type FuelWalletCtorParams = {
+    fuelWallet: WalletUnlocked;
+};
+
+export class FuelWalletClient extends WalletClientBase {
```

Comment:
I would make this an abstract class and create an implementation on top of it like with the solana and evm wallets: 
* https://github.com/goat-sdk/goat/blob/main/typescript/packages/wallets/solana/src/SolanaWalletClient.ts
* https://github.com/goat-sdk/goat/blob/main/typescript/packages/wallets/solana/src/SolanaKeypairWalletClient.ts

This allows you to have other wallet implementations in the future (e.g smart wallets, crossmint wallets, etc.) and have all plugins still work for them :)

Reply:
Ty!

https://github.com/goat-sdk/goat/pull/126/commits/e8b890bc7256ea904b3ca61477ade58146dcaf7c